### PR TITLE
Added Functional Test Cases

### PR DESCRIPTION
**Pull request by pgvishnuram**
* added basic test cases to verify cli option
* added test case to verify the path exists after cert generation
* added functional tests to create kuberntes cluster and run the certgenrator cron and validate result

### CODE REVIEW

**Comment by pgvishnuram** [comment]
@danielhoherd - pls add your review

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,12 +36,22 @@ jobs:
     steps:
       - trivy-scan:
           image_name: certgenerator
+  functional-test:
+    machine:
+      image: ubuntu-2004:202104-01
+      resource_class: xlarge
```
Comment (danielhoherd) [inline .circleci/config.yml:42]: This likely only needs a docker instance, not a full VM.
```suggestion
    docker:
      image: cimg/python:3.9
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -53,6 +63,27 @@ executors:
...
+        default: $CIRCLE_PROJECT_REPONAME
+    steps:
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: Run component test
+          command: |
+            set -xe
+            pyenv global 3.9.4
```
Comment (danielhoherd) [inline .circleci/config.yml:80]: If we use the `cimg/python:3.9` we won't need this line.
```suggestion
```

File: .circleci/config.yml (hunk 1 above)
Comment (danielhoherd) [reply .circleci/config.yml:42]: Oh, I see we are doing a full kind cluster. Interesting.

**Review by danielhoherd — APPROVED** [review]
Fantastic! I love the clean pytest_kind stuff, very cool! We need to take that as a cue for astronomer/astronomer testing and more.

File: .circleci/config.yml (hunk 1 above)
Comment (danielhoherd) [inline .circleci/config.yml:42]: Do you think we could get by with `large` ?
```suggestion
      resource_class: large
```

Comment (danielhoherd) [reply .circleci/config.yml:42]: FYI I tested this and large does work. https://app.circleci.com/pipelines/github/astronomer/certgenerator/12/workflows/46d1383d-6615-480f-ab75-7c6f62f14a27

**Pull request merged**